Cargo.toml

```diff
@@ -19,33 +19,20 @@ members = ["tests/wasm_bindgen"]
 
 [dependencies]
 log = { version = "0.4", optional = true }
+libc = "0.2.54"
```

Comment:
I don't think we want to depend on `libc` for all targets. Maybe instead construct one big `cfg` expression with enumeration of targets which use `libc`?

Reply:
Done, I just used `unix || wasi`. Although that's not perfect, as we don't use libc on all unix targets (only 7 of them), but writing all that out makes the `Cargo.toml` way harder to read.

Reply:
We also could remove `libc` dependency for WASI as well, `libc` [code](https://github.com/rust-lang/libc/blob/0e702c1b4e2e56788e8b67d3efd9c242807c3d4b/src/wasi.rs#L1269) is quite straightforward. Not sure if it's worth though.

Reply:
We totally could (Strictly speaking we could do it for any libc declaration), but I think it’s best to not duplicate declarations from the libc, if possible.

Reply:
If we wanted we could not depend on the libc at all for any target, by copying over the 6 or 7 libc declarations needed to make everything link. But I think that would make breakage/instability more likely, not less.

Reply:
It's just a bit strange for me to depend on `libc` crate for a target which, well, does not have `libc`.

Reply:
So I think they added it because [`wasi-libc` defines that symbol](https://github.com/CraneStation/wasi-libc/blob/320054e84f8f2440def3b1c8700cedb8fd697bf8/libc-bottom-half/headers/public/wasi/core.h#L708). However, it's just a macro that [links to the `wasi_unstable` module](https://github.com/CraneStation/wasi-libc/blob/94d34a13ad223930dce354ac33d3dfc73e9b5122/libc-bottom-half/headers/public/wasi/core.h#L451-L452). From their Github issues, it seems like the `wasi_unstable` module [is an implementation detail and may change in the future](https://github.com/CraneStation/wasmtime/issues/74#issuecomment-477779927). 

Given that, linking to the libc seems like the most stable approach, so we are not depending on an unstable components of the WASI API.

Reply:
In my understanding we shouldn't care about `wasi-libc` at all. WASI defines [Core API](https://github.com/CraneStation/wasmtime/blob/master/docs/WASI-api.md) and [`__wasi_random_get`](https://github.com/CraneStation/wasmtime/blob/master/docs/WASI-api.md#random_get) is part of it. So we should be able to use it directly without any issues.

cc @alexcrichton @gnzlbg